homeassistant/components/roon/__init__.py

```diff
@@ -0,0 +1,91 @@
+"""
+Roon (www.roonlabs.com) component.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/roon/
+"""
+import logging
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST
+from homeassistant.helpers import device_registry as dr
+
+# We need an import from .config_flow, without it .config_flow is never loaded.
+from .config_flow import configured_hosts
+from .const import CONF_CUSTOM_PLAY_ACTION, DATA_CONFIGS, DOMAIN
+from .server import RoonServer
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass, config):
+    """Set up the Roon platform."""
+    conf = config.get(DOMAIN)
+    if conf is None:
+        conf = {}
+
+    hass.data[DOMAIN] = {}
+    hass.data[DATA_CONFIGS] = {}
+    configured_servers = configured_hosts(hass)
+
+    if CONF_HOST not in conf:
+        # User has component configured but with no configured host
+        if not configured_servers:
+            # trigger config flow (can be removed once roon is added as official component to hass)
+            hass.async_create_task(
+                hass.config_entries.flow.async_init(
+                    DOMAIN, context={"source": config_entries.SOURCE_IMPORT}
+                )
+            )
+        return True
+    # setup component with config from configfile
+    host = conf[CONF_HOST]
+    # Store config in hass.data so the config entry can find it
+    hass.data[DATA_CONFIGS][host] = conf
+    # If configured, the server will be set up during config entry phase
+    if host in configured_servers:
+        return True
+    # No existing config entry found, try importing it or trigger link
+    # config flow if no existing auth. Because we're inside the setup of
+    # this component we'll have to use hass.async_add_job to avoid a
+    # deadlock: creating a config entry will set up the component but the
+    # setup would block till the entry is created!
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_IMPORT}, data=conf
+        )
+    )
+    return True
+
+
+async def async_setup_entry(hass, entry):
+    """Set up a roonserver from a config entry."""
+    host = entry.data[CONF_HOST]
+    config = hass.data[DATA_CONFIGS].get(host)
+    if config is None:
+        custom_play_action = None
+    else:
+        custom_play_action = config[CONF_CUSTOM_PLAY_ACTION]
+    roonserver = RoonServer(hass, entry, custom_play_action)
+
+    if not await roonserver.async_setup():
+        return False
+
+    hass.data[DOMAIN][host] = roonserver
```

Comment:
Let's store the server under the `config_entry.entry_id` key instead. That's standard and safer.
```suggestion
    hass.data[DOMAIN][entry.entry_id] = roonserver
```